packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -965,6 +967,16 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
             metadata,
         );
 
+        const shouldRunExpiry = this.runtimeOptions.gcOptions?.gcAllowed
+        && !this.runtimeOptions.gcOptions?.disableGC
+        && !this.runtimeOptions.gcOptions?.runSweep;
```

Comment:
I have a hard time reasoning about all 8 possible combinations here (2^3).
Can we please replace it with proper enum? I'd think there are only 3 states in this enum:
- disabled
- marking only
- mark & sweep

I also think you got the !runSweep wrong.